src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs

```diff
@@ -100,7 +100,15 @@ protected async Task<IActionResult> GetDetector(TResource resource, string detec
             }
 
             RuntimeContext<TResource> cxt = PrepareContext(resource, startTimeUtc, endTimeUtc, Form: form, detectorId: detectorId);
-            var detectorResponse = await GetDetectorInternal(detectorId, cxt);
+            Tuple<Response, List<DataProviderMetadata>> detectorResponse = null;
+            try
+            {
+                detectorResponse = await GetDetectorInternal(detectorId, cxt);
+            }
+            catch (Exception ex) when (ex.InnerException is ApiNotSupportedException)
+            {
+                return BadRequest(ex.Message);
+            }
```

Comment:
I think the detector API should still return server error and not client error as client cannot do anything to correct the request. The issue is in the server using an unsupported dependent API call

Reply:
Hey Shekhar, yeah I can see what you mean but don't you think it is the job of the detector authors who should be concerned with that? As their clients are the end users and our clients are the detector authors themselves so it shouldn't impact platform SLA.